modules/smart-agent_logstash/conf/07-queued_disk.yaml

```
@@ -0,0 +1,25 @@
+module: logstash
+name: queued disk
+
+transformation: ".mean(over='10m')"
+aggregation: true
+
+signals:
+  disk:
+    metric: node.stats.pipelines.queue.queue_size_in_bytes
+    rollup: delta
+  signal:
+    formula: (disk / 1000000)
```

instead of applying a static threshold check to the queue which, again, fully depends on each environment may by can you try to make this check more generic by making a comparison between `node.stats.pipelines.queue.max_queue_size_in_bytes` and `node.stats.pipelines.queue.queue_size_in_bytes`.

```suggestion
  size:
    metric: node.stats.pipelines.queue.queue_size_in_bytes
  max:
    metric: node.stats.pipelines.queue.max_queue_size_in_bytes
  signal:
    formula: (size/max).scale(100).fill(value=0).publish('signal')
```

from this way you can define relative percentage based threshold which could fit to everybody.

also, it seems redundant to the previous one event count detector. you can keep it but you should probably use `disabled: true` to disable it by default. this allow interested users to keep a hard queue limit count for who know it but don't enable alert by default.

---

Monitoring disk size in Logstash queue is different than the queued events in the pipeline. The disk size is cumulative of all pipelines so it is not quite the same. I'd like to keep this as static threshold.

---

ok it was only a suggestion to make this detector more generic. while most of your detectors use absolute thresholds they are not usable and useful out of the box for other users as seen above.

---

ok it was only a suggestion to make this detector more generic. while most of your detectors use absolute thresholds they are not usable and useful out of the box for other users as seen above.